pdf/pdfrenderthread.cpp

```diff
@@ -33,6 +33,72 @@ class PDFRenderThreadQueue;
 
 const QEvent::Type Event_JobPending = QEvent::Type(QEvent::User + 1);
 
+class SearchThread: public QThread
+{
+    Q_OBJECT
+public:
+    SearchThread(Poppler::Document *document, QObject *parent = 0)
+        : QThread(parent), m_document(document)
+    {
+    }
+    ~SearchThread()
+    {
+        requestInterruption();
+        wait();
+    }
+
+    QList<QPair<int, QRectF>> m_matches;
+
+    void start(const QString& search, uint startPage = 0)
+    {
+        requestInterruption();
+        wait();
+
+        m_search = search;
+        m_startPage = startPage;
+        QThread::start();
+    }
+
+    void run() {
+        m_matches.clear();
+        for (int i = 0; i < m_document->numPages() && !isInterruptionRequested(); ++i) {
+            int ipage = (m_startPage + i) % m_document->numPages();
+            Poppler::Page *page = m_document->page(ipage);
+
+            double sLeft, sTop, sRight, sBottom;
+            float scaleW = 1.f / page->pageSizeF().width();
+            float scaleH = 1.f / page->pageSizeF().height();
+            bool found;
+            found = page->search(m_search, sLeft, sTop, sRight, sBottom,
+                                 Poppler::Page::FromTop,
+                                 Poppler::Page::IgnoreCase);
+            while (found) {
```

Comment:
Could later consider if it'd make sense to report matches while search is not yet finished.